core/src/sumeragi/main_loop.rs

```diff
@@ -1007,11 +1000,12 @@ pub fn run<F>(
                         }
 
                         trace!("I, the proxy tail, have received a block.");
-                        for event in Vec::<Event>::from(&block) {
-                            trace!(?event);
-                            sumeragi.events_sender.send(event).unwrap_or(0);
+                        if sumeragi.events_sender.receiver_count() > 0 {
```

Comment:
ah, this is what we needed. The pattern is pervasive, see if you can extract it into a function

Reply:
I'm going to fix this separately in #2969.

Reply:
And backport a fix to LTS.